File: 0275-h-index-ii/0275-h-index-ii.py
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        n=len(citations)
        low,high,res=0,n-1,0

        while low<=high:
            guess=(low+high)//2

            if citations[guess]>=n-guess:
                res=n-guess
                high=guess-1
            else:
                low=guess+1
        
        return res

File: 0275-h-index-ii/0275-h-index-ii.cpp
class Solution {
public:
    int hIndex(vector<int>& citations) {
        int n=citations.size();
        int low=0;
        int high=n-1;
        int res=0;

        while(low<=high){
            int guess=(low+high)/2;

            if(citations[guess]>=n-guess){
                res=n-guess;
                high=guess-1;
            }
            else{
                low=guess+1;
            }
        }
        return res;
    }
};